File: algorithms/algorithms-java/src/main/java/com/lanceolata/coding/algorithms/java/leetcode/Question_0769_Max_Chunks_To_Make_Sorted.java
package com.lanceolata.coding.algorithms.java.leetcode;

public class Question_0769_Max_Chunks_To_Make_Sorted {
    public int maxChunksToSorted(int[] arr) {
        int res = 0, maxI = 0;
        for (int i = 0; i < arr.length; i++) {
            maxI = Math.max(maxI, arr[i]);
            if (maxI == i) {
                ++res;
            }
        }
        return res;
    }
}

File: algorithms/algorithms-cplusplus/leetcode/Question_0769_Max_Chunks_To_Make_Sorted.cc
#include <vector>

using namespace std;

class Solution {
 public:
  int maxChunksToSorted(vector<int>& arr) {
    int res = 0, maxI = 0;
    for (int i = 0; i < arr.size(); i++) {
      maxI = max(maxI, arr[i]);
      if (maxI == i) {
        ++res;
      }
    }
    return res;
  }
};
